File: Week 4/templates/D_bridges.py
# Input format: adj[i] is a list of the neighbors of node i
#               is_single[i][j] tells whether the edge from i to adj[i][j] is a single-lane bridge
#               adj[i].size() == is_single[i].size() for all i
# The input is 1-indexed
def bridges(n, adj, is_single):
    # Your code to find the minimum number of single-lane bridges goes here
    return -1 # This is a placeholder
    
n, m = map(int, input().split())
adj = [[] for _ in range(n + 1)]
is_single = [[] for _ in range(n + 1)]
for _ in range(m):
    u, v, is_single_lane = map(int, input().split())
    adj[u].append(v)
    is_single[u].append(is_single_lane == 1)
    adj[v].append(u)
    is_single[v].append(is_single_lane == 1)

print(bridges(n, adj, is_single))

File: Week 4/templates/D_bridges.cpp
#include <bits/stdc++.h>
using namespace std;

// Input format: adj[i] is a vector of the neighbors of node i
//               is_single[i][j] tells whether the edge from i to adj[i][j] is a single-lane bridge
//               adj[i].size() == is_single[i].size() for all i
// The input is 1-indexed
int bridges(int n, vector<vector<int>> adj, vector<vector<bool>> is_single){
    // Your code goes here
    return -1; // This is a placeholder
}

int main(){
    int n,m;
    cin >> n >> m;
    vector<vector<int>> adj(n + 1);
    vector<vector<bool>> is_single(n + 1);
    for(int i = 0; i < m; i++){
        int u,v,b;
        cin >> u >> v >> b;
        adj[u].push_back(v);
        adj[v].push_back(u);
        is_single[u].push_back(b);
        is_single[v].push_back(b);
    }
    cout << bridges(n, adj, is_single) << endl;
}